File: Sorting Algorithms/QuickSort.java
public class QuickSort{

    public static int partition(int array[],int low,int high){

        int pivot=array[high];
        int i=low-1;

        for(int j=low;j<high;j++){
            if(array[j]<pivot){
                i++;
                int temp=array[i];
                array[i]=array[j];
                array[j]=temp;
            }
        }
        int temp=array[i+1];
        array[i+1]=array[high];
        array[high]=temp;

        return i+1;
    }
    public static void quickSort(int array[],int low,int high){

        if(low<high){

            int pivot=partition(array,low,high);

            quickSort(array, low, pivot-1);
            quickSort(array, pivot+1,high);
        }
        
    }
    public static void main(String args[]) {

        int array[] = new int[] { 9, 8, 7, 6, 4, 3, 2, 11, 123, 0 };

        quickSort(array,0,array.length-1);

        for (int i = 0; i < array.length; i++)
            System.out.print(array[i] + " ");
        System.out.println();
    }
    
}

File: Sorting Algorithms/QuickSort.c
#include <stdio.h>
typedef long long int lld;

void swap(lld *a,lld *b){
    lld temp=*a;
    *a=*b;
    *b=temp;
}

// choose left most element as pivot
lld partition(lld arr[],lld low,lld high){
    
    // take last element as pivot 
    lld pivot=arr[high];
    lld i=low-1;
    // traverse from left to right
    // elements less tha pivot gets swaped with 'i'
    // finally pivot replaced with 'i' as left of pivot is lower and right is higher
    for(lld j=low;j<=high;j++){
        if(arr[j]<pivot){
            //takes all smaller element and place it to left
            i++;
            swap(&arr[j],&arr[i]);
        }
    }
    //change pivot to center
    swap(&arr[i+1],&arr[high]);
    //return pivot potision   
    return i+1;
}


void quick_sort(lld arr[],lld low,lld high){
    
    if(low<high){
        lld pivot=partition(arr,low,high);
        
        quick_sort(arr,low,pivot-1);
        quick_sort(arr,pivot+1,high);
    }
}

int main()
{
    lld arr[1000],n;
    scanf("%lld",&n);
 
    // get input   
    for(lld i=0;i<n;i++)
        scanf("%lld",&arr[i]);
    quick_sort(arr,0,n-1);
    for(lld i=0;i<n;i++)
        printf("%lld ",arr[i]);
    printf("\n");
    return 0;
}
